AbstractPyth.sol

```diff
@@ -59,4 +59,22 @@ abstract contract AbstractPyth is IPyth {
             return y - x;
         }
     }
+
+    // Changing modifier to public to be able to call it locally.
+    function updatePriceFeeds(bytes[] memory updateData) public virtual payable override;
```

Comment:
I think if you make this `public` it will use more gas than `external` because it will copy the `updateData` to the stack (?) I'm no EVM expert though so i could be wrong here. 

also if you change this, do you need to change the access modifier of the corresponding `IPyth` function below as well?

Reply:
updateData is a memory variable and memory to memory is a reference copy ([link](https://docs.soliditylang.org/en/v0.8.6/types.html#data-location-and-assignment-behaviour)).

Also some notes on public vs external.
External functions are the ones that can be called externally (a cross contract call) and public ones are external in addition to allowing local calls. It's possible to call same contract external function but it will be a cross contract call (with `this.method()`) and has more gas cost + sender/value will be modified.

All interface methods should be external (no public is allowed). Overriding an external to public is ok.